internal/proxyservice/tcp/mysql/plugin.go

```diff
@@ -0,0 +1,34 @@
+package mysql
+
+import (
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
+)
+
+type mysqlPlugin struct{}
+
+// NewConnector is required method on the tcp.Plugin interface. It returns a
+// tcp.Connector.
+//
+// The single argument passed in is of type connector.Resources. It contains
+// connector-specific config and a logger.
+func (examplePlugin *mysqlPlugin) NewConnector(_ connector.Resources) tcp.Connector {
+	return (&handler{}).ConnectorFunc
+}
+
+// PluginInfo is required as part of the Secretless plugin spec. It provides
+// important metadata about the plugin.
+func PluginInfo() map[string]string {
+	return map[string]string{
+		"pluginAPIVersion": "0.1.0",
+		"type":             "connector.tcp",
+		"id":               "mysql",
+		"description":      "processes connection requests to a MySQL database",
+	}
+}
+
+// GetTCPPlugin is required as part of the Secretless plugin spec for TCP connector
+// plugins. It returns the TCP plugin.
+func GetTCPPlugin() tcp.Plugin {
+	return &mysqlPlugin{}
+}
```

Comment:
Not something you need to do, but....

There's a cool trick used  in the stdlib to get rid of the empty `mysqlPlugin{}` struct which only exists to meet syntactic needs.  You can instead do:

```go
func NewConnector(_ connector.Resources) tcp.Connector {
	return (&handler{}).ConnectorFunc
}

type Plugin func (connector.Resources) tcp.Connector

func (p *Plugin) NewConnector(cr connector.Resources) tcp.Connector {
  return p(cr)
}
```

Then you'd do:

```go
func GetTCPPlugin() tcp.Plugin {
	return Plugin(NewConnector)
}
```
That's untested but you get the idea.  This is often useful when you're implementing 1 method interfaces.

Reply:
This is a neat pattern. Just needed a single change:
```
func (p Plugin) NewConnector(cr connector.Resources) tcp.Connector {
```